src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/INuGetTelemetryCollector.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Common;
+
+namespace NuGet.VisualStudio.Telemetry
+{
+    // This one used for emitting aggregated telemetry at VS solution close or VS instance close.
+    public interface INuGetTelemetryCollector
```

Comment:
Instead of a telemetry collector, consider a service that actually just collects data. Similar to what @zivkan suggested in https://github.com/NuGet/Client.Engineering/pull/579.

Reply:
@zivkan 
If I understand correctly this spec is not implemented yet, right?
Do you already have any mock up code to start? Not sure how to use ETW for this purpose.

Reply:
You don't have to use ETW, just think about the architecture in a similar way. 

Decouple data & telemetry sending.